orangecontrib/text/widgets/owimporttexts.py

```diff
@@ -0,0 +1,776 @@
+            clicked=browseaction.trigger
+        )
+        reloadbutton = QPushButton(
+            reloadaction.iconText(),
+            icon=reloadaction.icon(),
+            clicked=reloadaction.trigger,
+            default=True,
+        )
+
+        hbox.layout().addWidget(self.recent_cb)
+        hbox.layout().addWidget(browsebutton)
+        hbox.layout().addWidget(reloadbutton)
+
+        self.addActions([browseaction, reloadaction])
+
+        reloadaction.changed.connect(
+            lambda: reloadbutton.setEnabled(reloadaction.isEnabled())
+        )
+        box = gui.vBox(vbox, "Info")
+        self.infostack = QStackedWidget()
+
+        self.info_area = QLabel(
+            text="No text set selected",
+            wordWrap=True
+        )
+        self.progress_widget = QProgressBar(
+            minimum=0, maximum=0
+        )
+        self.cancel_button = QPushButton(
+            "Cancel", icon=icons.cancel_icon,
+        )
+        self.cancel_button.clicked.connect(self.cancel)
+
+        w = QWidget()
+        vlayout = QVBoxLayout()
+        vlayout.setContentsMargins(0, 0, 0, 0)
+        hlayout = QHBoxLayout()
+        hlayout.setContentsMargins(0, 0, 0, 0)
+
+        hlayout.addWidget(self.progress_widget)
+        hlayout.addWidget(self.cancel_button)
+        vlayout.addLayout(hlayout)
+
+        self.pathlabel = TextLabel()
+        self.pathlabel.setTextElideMode(Qt.ElideMiddle)
+        self.pathlabel.setAttribute(Qt.WA_MacSmallSize)
+
+        vlayout.addWidget(self.pathlabel)
+        w.setLayout(vlayout)
+
+        self.infostack.addWidget(self.info_area)
+        self.infostack.addWidget(w)
+
+        box.layout().addWidget(self.infostack)
+
+        self.__initRecentItemsModel()
+        self.__invalidated = True
+        self.__executor = ThreadExecutor(self)
+
+        QApplication.postEvent(self, QEvent(RuntimeEvent.Init))
+
+    def __initRecentItemsModel(self):
+        if self.currentPath is not None and \
+                not os.path.isdir(self.currentPath):
+            self.currentPath = None
+
+        recent_paths = []
+        for item in self.recent_paths:
+            if os.path.isdir(item.abspath):
+                recent_paths.append(item)
+        recent_paths = recent_paths[:OWImportTexts.MaxRecentItems]
+        recent_model = self.recent_cb.model()
+        for pathitem in recent_paths:
+            item = RecentPath_asqstandarditem(pathitem)
+            recent_model.appendRow(item)
+
+        self.recent_paths = recent_paths
+
+        if self.currentPath is not None and \
+                os.path.isdir(self.currentPath) and self.recent_paths and \
+                os.path.samefile(self.currentPath, self.recent_paths[0].abspath):
+            self.recent_cb.setCurrentIndex(0)
+        else:
+            self.currentPath = None
+            self.recent_cb.setCurrentIndex(-1)
+        self.__actions.reload.setEnabled(self.currentPath is not None)
+
+    def customEvent(self, event):
```

Comment:
IIUC this custom event handler, as used, prevents running the `start()` method's code as part of widget constructor. Like if `QTimer.singleShot(1, ...)` were used.